src/Core/Vipr.Core/CodeModel/OdcmProjection.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Vipr.Core.CodeModel.Vocabularies;
+
+namespace Vipr.Core.CodeModel
+{
+    public class OdcmProjection
+    {
+        public OdcmType Type { get; set; }
+
+        public OdcmCapabilities Capabilities { get; set; }
+
+        public OdcmCapabilities EnsureCapabilities()
```

Comment:
Agreed. Can't we capture this in the constructor or, if laziness is the intent, in the getter?